### PR TITLE
Adds some contributing guidelines

### PR DESCRIPTION
**Pull request by Cyberboss**
Adds a rule about how to specify string keys in an associated list.

Removes some stray = nulls that no longer match our guideline

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -274,7 +274,13 @@ Math operators like +, -, /, *, etc are up in the air, just choose which version
...
 * Bitwise AND - '&'
 	* Should be written as ```bitfield & bitflag``` NEVER ```bitflag & bitfield```, both are valid, but the latter is confusing and nonstandard.
-
+* Do not explicitly assign `null` to newly declared variables as this is already their default value
+* When defining new datums, place newly defined variables at the bottom
+* If a datum has already been declared somewhere, additional var definitions should not be made outside of that block
+* Associated lists declarations must have their key value quoted if it's a string
+	* WRONG: list(a = "b")
+	* RIGHT: list("a" = "b")
+* Do not call procs for reasons other than their intended purpose. (i.e. New(), Destroy(), Crossed(), Topic(), attackby(), attack_hand() should not be called directly). Create seperate procs that encapsule the functionality you're trying to perform instead.
```
Comment (RemieRichards) [inline .github/CONTRIBUTING.md]: Disagree ENTIRELY for ```attack_hand()```, ```attack_x()``` (paw/alien/etc.) and ```attackby()```

Comment (RemieRichards) [inline .github/CONTRIBUTING.md]: This last part is completely off-tone for the rest of this PR, remove it and you can choose to apply it where you wish, it doesn't deserve codifying in the guidelines because there are a multitude of acceptable uses.

Comment (Cyberboss) [inline .github/CONTRIBUTING.md]: I'll argue that there's always 2 better ways to to it

Case 1: You want some sort of special effect from attack* to trigger. In that case it's better to have a common proc and call it from inside attackby and the place you were going to call attackby.

Case 2: You actually want to trigger an attack on the item. For that we have melee_item_attack_chain() which handles the process correctly.

Considering I want to start enforcing attackby() and attack_hand() start `. = ..()`ing (because there's some stuff that needs to go in the parent proc) having direct calls like these can lead to unintended side effects

Comment (lzimann) [inline .github/CONTRIBUTING.md]: I'm with Remie in this one.

Comment (Cyberboss) [inline .github/CONTRIBUTING.md]: @ChangelingRain? @optimumtact?

Comment (Tacolizard) [inline .github/CONTRIBUTING.md]: hey later on in the file there's a code trick thing tutorial that uses = null. https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#istypeless-for-loops

Comment (MrPerson) [inline .github/CONTRIBUTING.md]: Either way we need to pick a strategy and start suggesting/forcing it everywhere.

I'm of the opinion that if we're going to repeat parts of a chain (ie call attack_hand() etc), really you should be starting as high as possible up the chain.

Comment (tortellinitony) [inline .github/CONTRIBUTING.md]: Yea this has nothing to do with the rest of the PR. Obviously, you're not one to describe what your PRs are about, but this clearly is centered around defining variables. This is a different issue and should be discussed in a different PR.

Comment (optimumtact) [inline .github/CONTRIBUTING.md]: these procs are usually so specific they dont' get called out of their own usual attack chain anyway (save cases where things like attack_paw call attack_hand) so it's irrelevant either way.

99% of the time it's right to break out whatever behaviour want into a subproc and call it in both anyway.

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -274,7 +274,13 @@ Math operators like +, -, /, *, etc are up in the air, just choose which version
 #### Use
 * Bitwise AND - '&'
 	* Should be written as ```bitfield & bitflag``` NEVER ```bitflag & bitfield```, both are valid, but the latter is confusing and nonstandard.
-
+* Do not explicitly assign `null` to newly declared variables as this is already their default value
```
Comment (Fox-McCloud) [inline .github/CONTRIBUTING.md]: Strongly disagree with this. 

Yeah, it is their default state, but it's clearer when it's explicitly null. It doesn't make things look ugly and it helps cement it in people's mind that it's a holder.

It's far easier to glance at an datum/turf/mob/object and see what the holders are by `= null`. It's especially useful for tracking down other references, very quickly.

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -274,7 +274,13 @@ Math operators like +, -, /, *, etc are up in the air, just choose which version
...
+* When defining new datums, place newly defined variables at the bottom
```
Comment (Fox-McCloud) [inline .github/CONTRIBUTING.md]: I'd say this should have an exception to it.

There's some instances where having a var that is newly declared be with previously declared ones.

for instance, if there's an object that has a lot of different icon states, and uses something like `base_icon`, it's *far* more readable to have that variable with `icon` and `icon_state` than off by itself at the bottom.

Generally a good rule, but I don't think it should be adhered to blindly.

File: .github/CONTRIBUTING.md [hunk 4]
```diff
@@ -274,7 +274,13 @@ Math operators like +, -, /, *, etc are up in the air, just choose which version
...
+* If a datum has already been declared somewhere, additional var definitions should not be made outside of that block
```
Comment (Fox-McCloud) [inline .github/CONTRIBUTING.md]: Again, I think this is a generally "ok" thing for most things, but other times, it's going to be better to just have the var defined on the module that has all the behaviors related to that var.

**Comment by Tacolizard** [comment]
no! please revert https://github.com/tgstation/tgstation/pull/29345/commits/abc5c89c7beaab6a672b26bc28b3f633e03ba036

**Comment by MrStonedOne** [comment]
https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md

>You are going to be expected to document all your changes in the pull request
